compaction/time_window_compaction_strategy.cc

```diff
@@ -296,9 +296,14 @@ time_window_compaction_strategy::get_reshaping_job(std::vector<shared_sstable> i
             // When trimming, let's keep sstables with overlapping time window, so as to reduce write amplification.
             // For example, if there are N sstables spanning window W, where N <= 32, then we can produce all data for W
             // in a single compaction round, removing the need to later compact W to reduce its number of files.
-            boost::partial_sort(multi_window, multi_window.begin() + max_sstables, [](const shared_sstable &a, const shared_sstable &b) {
+            auto cmp = [] (const shared_sstable &a, const shared_sstable &b) {
                 return a->get_stats_metadata().max_timestamp < b->get_stats_metadata().max_timestamp;
-            });
+            };
+            if (multi_window.size() > max_sstables) {
+                std::ranges::partial_sort(multi_window, multi_window.begin() + max_sstables, cmp);
+            } else {
```

Comment:
can we avoid the conditional with: 
multi_window.begin() + std::min(max_sstables, multi_window.size()) 
?

Reply:
other than that, looks good to me

Reply:
btw, I suggest it for readability, not as an optimization since min() is a conditional under the hood too.